src/CheckForSafeFunctionCall.ts

```diff
@@ -1,101 +1,136 @@
+import {
+    Loop,
+    Joinpoint,
+    If,
+    ArrayAccess,
+    Continue,
+    MemberAccess,
+    Varref,
+    Call,
+    FileJp,
+    FunctionJp,
+    Body,
+} from "@specs-feup/clava/api/Joinpoints.js";
 
-    select file.function.body.arrayAccess end
-    apply
-        if($arrayAccess.use.indexOf('write') === -1) {
-            continue;
+/**************************************************************
+ *
+ *                       checkForSafeFunctionCall
+ *
+ **************************************************************/
+export function CheckForSafeFunctionCall(
+    $function: Function,
+    $arrayAccess: ArrayAccess,
+    $memberAccess: MemberAccess,
+    $varref: Varref,
+    $file: FileJp
+) {
+    const new_safefunctionCallslist: string[] = [];
+    for (const $function of Query.search(FileJp).search(FunctionJp)) {
+        if ($function.params.length > 0) {
+            if (new_safefunctionCallslist.indexOf($function.name) == -1)
+                new_safefunctionCallslist.push($function.name);
         }
-        
-        var currentRegion = $arrayAccess.arrayVar.getDescendantsAndSelf('varref')[0].vardecl.currentRegion;
-        if((currentRegion !== undefined && currentRegion.joinPointType === 'file')
-            || $arrayAccess.arrayVar.getDescendantsAndSelf('varref')[0].vardecl.isParam) {
-            
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);
+    }
+    for (const $function of Query.search(FileJp)
+        .search(FunctionJp)
+        .search(Call)) {
+        if (new_safefunctionCallslist.indexOf($function.name) != -1) {
+            new_safefunctionCallslist.splice(
+                new_safefunctionCallslist.indexOf($function.name),
+                1
+            );
         }
-        
-/*
-        if (
-            $arrayAccess.use.indexOf('write') !== -1 && 
-                (
-                    $arrayAccess.arrayVar.getDescendantsAndSelf('varref')[0].vardecl.currentRegion.joinPointType === 'file'
-                    ||
-                    $arrayAccess.arrayVar.getDescendantsAndSelf('varref')[0].vardecl.isParam
-                )
-            )
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);
-*/			
-    end
-    condition new_safefunctionCallslist.indexOf($function.name) !== -1 end
+    }
 
-    select file.function.body.memberAccess end
-    apply
-        if($memberAccess.use.indexOf('write') === -1) {
-            continue;
-        }
-        
-        var currentRegion = $memberAccess.getDescendantsAndSelf('varref')[0].vardecl.currentRegion;
-        if(currentRegion !== undefined && currentRegion.joinPointType === 'file') {
-            new_safefunctionCallslist.splice(new_safefunctionCallslist.indexOf($function.name),1);
+    for (const $function of Query.search(FileJp)
+        .search(FunctionJp)
+        .search(Body)
+        .search(ArrayAccess)) {
```

Comment:
```suggestion
    for (const $function of Query.search(FileJp).search(FunctionJp)) {
        for (const $arrayAccess of Query.searchFrom(
            $function.body,
            ArrayAccess
        )) {
```

Depois precisas de um `}` no fim.